### PR TITLE
Add the foundation for the template studio

### PR DESCRIPTION
**Pull request by m-vo**
This PR does three things in preparation for the new template studio (that can be reviewed individually by going through the commits):

* Add a `backend_icon` Twig filter as a Twig-equivalent for `Image::getHtml()``.
* Add a tab component for the backend, that allows to dynamically add/remove panels by streaming content into a designated container.
* Add a backend controller and basic layout and styling for the template studio. To keep things manageable, In this PR, only the identifier tree is rendered and tabs get opened on click - no editors or other content. For this reason, the backend menu item is also not added, yet. To test, open `/contao/template-studio`.

### CODE REVIEW

File: core-bundle/src/Twig/Runtime/BackendHelperRuntime.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public function __construct(private readonly ContaoFramework $framework)
+    {
+    }
+
+    public function icon(string $src, string $alt = '', HtmlAttributes|null $attributes = null): string
+    {
+        $image = $this->framework->getAdapter(Image::class);
+
+        return $image->getHtml($src, $alt, !$attributes ? '' : $attributes->toString(false));
```
Comment (leofeyer) [inline core-bundle/src/Twig/Runtime/BackendHelperRuntime.php]:
```suggestion
        return $image->getHtml($src, $alt, $attributes ? $attributes->toString(false) : '');
```

Just a nitpick

File: core-bundle/src/Twig/Runtime/BackendHelperRuntime.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public function __construct(private readonly ContaoFramework $framework)
+    {
+    }
+
+    public function icon(string $src, string $alt = '', HtmlAttributes|null $attributes = null): string
+    {
+        $image = $this->framework->getAdapter(Image::class);
+
+        return $image->getHtml($src, $alt, $attributes ? $attributes->toString(false)) : '';
```
Comment (m-vo) [inline core-bundle/src/Twig/Runtime/BackendHelperRuntime.php]:
```suggestion
        return $image->getHtml($src, $alt, $attributes ? $attributes->toString(false)) : '');
```

File: core-bundle/src/Twig/Runtime/BackendHelperRuntime.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public function __construct(private readonly ContaoFramework $framework)
+    {
+    }
+
+    public function icon(string $src, string $alt = '', HtmlAttributes|null $attributes = null): string
+    {
+        $image = $this->framework->getAdapter(Image::class);
+
+        return $image->getHtml($src, $alt, $attributes ? $attributes->toString(false)) : '');
```
Comment (leofeyer) [inline core-bundle/src/Twig/Runtime/BackendHelperRuntime.php]:
```suggestion
        return $image->getHtml($src, $alt, $attributes ? $attributes->toString(false) : '');
```

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @m-vo.

**Pull request merged**